streams/src/main/java/org/apache/kafka/streams/state/internals/InMemorySessionStore.java

```diff
@@ -0,0 +1,376 @@
+            return null;
+        }
+
+        final ConcurrentNavigableMap<Bytes, ConcurrentNavigableMap<Long, byte[]>> keyMap = endTimeMap.get(endTime);
+        if (keyMap != null) {
+            final ConcurrentNavigableMap<Long, byte[]> startTimeMap = keyMap.get(key);
+            if (startTimeMap != null) {
+                return startTimeMap.get(startTime);
+            }
+        }
+        return null;
+    }
+
+    @Deprecated
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> findSessions(final Bytes key,
+                                                                  final long earliestSessionEndTime,
+                                                                  final long latestSessionStartTime) {
+        removeExpiredSegments();
+
+        return new InMemorySessionStoreIterator(key,
+                                                key,
+                                                latestSessionStartTime,
+                                                endTimeMap.tailMap(earliestSessionEndTime, true).entrySet().iterator());
+    }
+
+    @Deprecated
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> findSessions(final Bytes keyFrom,
+                                                                  final Bytes keyTo,
+                                                                  final long earliestSessionEndTime,
+                                                                  final long latestSessionStartTime) {
+        removeExpiredSegments();
+
+        if (keyFrom.compareTo(keyTo) > 0) {
+            LOG.warn("Returning empty iterator for fetch with invalid key range: from > to. "
+                + "This may be due to serdes that don't preserve ordering when lexicographically comparing the serialized bytes. " +
+                "Note that the built-in numerical serdes do not follow this for negative numbers");
+            return KeyValueIterators.emptyIterator();
+        }
+
+        return new InMemorySessionStoreIterator(keyFrom,
+                                                keyTo,
+                                                latestSessionStartTime,
+                                                endTimeMap.tailMap(earliestSessionEndTime, true).entrySet().iterator());
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetch(final Bytes key) {
+        removeExpiredSegments();
+
+        return new InMemorySessionStoreIterator(key, key, Long.MAX_VALUE, endTimeMap.entrySet().iterator());
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetch(final Bytes from, final Bytes to) {
+        removeExpiredSegments();
+
+        return new InMemorySessionStoreIterator(from, to, Long.MAX_VALUE, endTimeMap.entrySet().iterator());
+    }
+
+    @Override
+    public boolean persistent() {
+        return false;
+    }
+
+    @Override
+    public boolean isOpen() {
+        return this.open;
```

Comment:
nit: unnecessary qualifier
```suggestion
        return open;
```